sdk/python/kfp_tekton/compiler/compiler.py

```diff
@@ -136,38 +158,30 @@ def _group_to_dag_template(self, group, inputs, outputs, dependencies):
       },
       'spec': {}
     }
-  
-    # Generate inputs section.
-    if inputs.get(group.name, None):
-      template_params = [{'name': x[1] if x[1] else x[0]} for x in inputs[group.name]]
-      template['spec']['params'] = template_params
       
     # Generates template sections unique to conditions
     if isinstance(sub_group, dsl.OpsGroup) and sub_group.type == 'condition':
       subgroup_inputs = inputs.get(sub_group.name, [])
       condition = sub_group.condition
 
-      operand1_value = str(condition.operand1)
-      operand2_value = str(condition.operand2) 
+      operand1_value = self._resolve_value_or_reference(condition.operand1, subgroup_inputs)
+      operand2_value = self._resolve_value_or_reference(condition.operand2, subgroup_inputs)
 
-      # Generates the operand values and exits gracefully if task outputs are trying to be used
-      # TODO this can be removed when Conditions support parameter passing from task outputs at which point
-      # the _resolve_value_or_reference method from kfp.Compiler can be used instead
+      # Adds params to condition template
+      params = []
       if isinstance(condition.operand1, dsl.PipelineParam):
-        parameter_name = self._pipelineparam_full_name(condition.operand1)
-        task_names = [task_name for param_name, task_name in subgroup_inputs if param_name == parameter_name]
-        if task_names[0]:
-          raise TypeError("Conditions do not currently support parameter passing from task outputs")
-        operand1_value =  '$(params.'+parameter_name+')'
+        params.append({'name': operand1_value[9: len(operand1_value) - 1]}) # Substring to remove parameter reference wrapping
       if isinstance(condition.operand2, dsl.PipelineParam):
-        parameter_name = self._pipelineparam_full_name(condition.operand2)
-        task_names = [task_name for param_name, task_name in subgroup_inputs if param_name == parameter_name]
-        if task_names[0]:
-          raise TypeError("Conditions do not currently support parameter passing from task outputs")
```

Comment:
Since the parameter passing only works on master and future Tekton 0.12 release. Maybe we should add a warning messaging using the [logging](https://docs.python.org/3/howto/logging.html) library.